### ISSUE TITLE
Error parsing header values

### ISSUE THREAD

**Issue by dfirsht** [IBM-Swift/Kitura#744]
In an incoming request, the header values seem to have been merged together.
<img width="878" alt="screen shot 2016-09-16 at 2 43 23 pm" src="https://cloud.githubusercontent.com/assets/5191832/18599056/f8cc6014-7c1b-11e6-81ce-c7435ecee21f.png">

Here is the raw request, (as captured by Charles)
```
POST /sentcard HTTP/1.1
Host: localhost:9080
Content-Type: application/json
Connection: keep-alive
Accept: */*
User-Agent: MessagesExtension/1 CFNetwork/808.0.2 Darwin/15.6.0
Content-Length: 58
Accept-Language: en-us
Accept-Encoding: gzip, deflate

{"id":"123","card_type":"oneDay","location_id":"USWA0395"}
```

Note: This issue is somewhat intermediate and about 20% of requests are error free.
Other requests though get the error `Failed to parse a request. Parsed fewer bytes than were passed to the HTTP parser `

**Comment by shmuelk** [comment]
@dfirsht Is possible to get a copy of your server and the client driving the requests?

**Comment by dfirsht** [comment]
@shmuelk PMed it.

**Comment by shmuelk** [comment]
@dfirsht Thanks for sending the test case. It took a bit of extending the test case to recreate the failure (I added a loop). But in the end I got it to fail.

While this isn't a duplicate of issue #734, the fix being put together for issue #734 fixed your problem. In my test case I called the post routine forty times in a loop, with no failures, using the code from fix of issue #734.

The fix for issue #734 is still work in progress. I hope to finish it today.
